### PR TITLE
split update_index_and_logfile.py into 3 scripts, and use Dask

### PR DESCRIPTION
**Pull request by miseminger**
new scripts are:

- `gvf2indexandlog.py`: takes one GVF, creates two TSVs: one index, one logfile.
- `merge_indices.py`: reads all index files from `gvf2indexandlog.py`, as well as an optional original mutation index, and merges them into a new index using Dask.
- `merge_logfiles.py`: reads all log files from `gvf2indexandlog.py`, as well as a log header file, and an optional original mutation index, and merges them into a new log using Dask. It also prints out the number of mutations in the log that have 5 or more corresponding lineages.

the old script, `update_index_and_logfile.py`, is obsolete after the addition of these three scripts, and can be deleted.

also important to note: the mutation index columns have changed since we last made an index. The new columns are:
`['pos', 'mutation', 'hgvs_aa_mutation', 'hgvs_nt_mutation', 'gene', 'protein_name', 'alias', 'hgvs_alias', 'alias_protein', 'Pokay_annotation', 'lineages']`

### CODE REVIEW

**Comment by miseminger** [comment]
@miseminger tested with Dask version 2021.10.0. Will update to the current version and check that it still works.

File: bin/merge_indices.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    # read all indices into dask df
+    ddf = dd.read_csv(indices_to_merge, sep='\t') 
+    # fillna to make groupby() work
+    ddf = ddf.fillna('n/a')
+    # specify which columns to group by
+    group_cols = [x for x in ddf.columns if x!='lineages']
+    '''
+    # specify meta df: this throws an error, but the inferred meta works fine
+    schema = {**dict.fromkeys(ddf.columns, 'str'), 'pos':'int'}
+    meta_df = pd.DataFrame(columns=schema.keys()).astype(schema)
```
Comment (anwarMZ) [inline bin/merge_indices.py:55]: This specific line uses pandas and if I am not wrong, pandas are not used anywhere else in this script. Is it possible to convert pandas to dask here too? This should help us with containerization. 
The current version of dask is [2023.10.1-py11-ol9_cv1](https://hub.docker.com/layers/biocontainers/dask/2023.10.1-py11-ol9_cv1/images/sha256-8bd5e6fad3a0c3e8009edab4217662b8c5b56bc11a8d2305f3b707509299dd71?context=explore). Can you please try with this version?

File: bin/merge_logfiles.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    # sort by 'pos'
+    ddf = ddf.sort_values("pos")
+
+    # if a mutation index is provided, remove
+    # previously-identified mutations from the logfile
+    if original_index!=None:
+
+        # convert original_index to match the logfile columns
+        # read index into pandas df
+        og_index = pd.read_csv(original_index, sep='\t').fillna('n/a')
```
Comment (anwarMZ) [inline bin/merge_logfiles.py:64]: replace pandas with dask here too

File: bin/merge_logfiles.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        ddf = ddf[['pos', 'new_mutations', 'lineages']]
+        ddf['pos'] = ddf['pos'].astype(int)
+        
+    # print number of new mutations in >=5 lineages
+    num_lineages = ddf['lineages'].compute()
+    num_lineages = num_lineages.str.split(',').str.len()
+    print("New mutations in >=5 lineages: ", (num_lineages>=5).sum())
+    
+    # add log header
+    log_header_df = pd.read_csv(log_header, sep='\t', names=["pos", "new_mutations", "lineages"])
```
Comment (anwarMZ) [inline bin/merge_logfiles.py:97]: replace pandas with dask here too

**Review by anwarMZ — CHANGES_REQUESTED** [review]
Please replace the minimal use of pandas with dask in `bin/merge_indices.py` and  `bin/merge_logfiles.py` scripts and test it on the latest dask version [2023.10.1-py11-ol9_cv1](https://hub.docker.com/layers/biocontainers/dask/2023.10.1-py11-ol9_cv1/images/sha256-8bd5e6fad3a0c3e8009edab4217662b8c5b56bc11a8d2305f3b707509299dd71?context=explore)

**Comment by miseminger** [comment]
> Please replace the minimal use of pandas with dask in `bin/merge_indices.py` and `bin/merge_logfiles.py` scripts and test it on the latest dask version [2023.10.1-py11-ol9_cv1](https://hub.docker.com/layers/biocontainers/dask/2023.10.1-py11-ol9_cv1/images/sha256-8bd5e6fad3a0c3e8009edab4217662b8c5b56bc11a8d2305f3b707509299dd71?context=explore)

Latest Dask version (2024.2.1) depends on Pandas 2.2.1; can we update Pandas in `environment.yml`?

File: bin/merge_indices.py (hunk 1 above)
Comment (miseminger) [reply bin/merge_indices.py]: deleted the use of `meta` here and it works fine (though would be nice to specify someday)

File: bin/merge_logfiles.py (hunk 2 above)
Comment (miseminger) [reply bin/merge_logfiles.py]: done

**Review by anwarMZ — APPROVED** [review]

**Pull request merged**